runtime/altair/src/migrations.rs

```diff
@@ -268,29 +481,21 @@ mod orml_tokens_migration {
 			let mut migrated_entries: u64 = 0;
 
 			// Burn all AUSD tokens under the old CurrencyId and mint them under the new one
-			orml_tokens::Accounts::<Runtime>::iter()
+			Accounts::<T>::iter()
 				.filter(|(_, old_currency_id, _)| *old_currency_id == DEPRECATED_AUSD_CURRENCY_ID)
 				.for_each(|(account, _, account_data)| {
+					log::info!(
+						"{LOG_PREFIX} Migrating account with balance: {}",
+						account_data.free
+					);
+
 					let balance = account_data.free;
-					// Burn the amount under the old, hardcoded CurrencyId
-					<orml_tokens::Pallet<Runtime> as Mutate<AccountId>>::burn_from(
-						DEPRECATED_AUSD_CURRENCY_ID,
-						&account,
-						balance,
-						Precision::Exact,
-						Fortitude::Force,
-					)
-					.map_err(|e| {
-						log::error!(
-							"Failed to call burn_from({:?}, {:?}, {balance}): {:?}",
-							DEPRECATED_AUSD_CURRENCY_ID,
-							account,
-							e
-						)
-					})
-					.ok();
+
+					// Remove account data and reduce total issuance manually at the end
+					Accounts::<T>::remove(account.clone(), DEPRECATED_AUSD_CURRENCY_ID);
```

Comment:
Better than burning 👍🏻

Reply:
Here, this approach is a bit more lightweight. Moreover, burning was not possible without many "hacks" because the runtime does not know the deprecated currency variant anymore. For storages, we can mitigate this by declaring a storage alias. However, abstraction layers such as minting or burning are not affected by this.

Reply:
And we also have free resources for an account that will never be used, so great!